src/cursor/cur_version.c

```diff
@@ -545,12 +551,13 @@ __curversion_next_single_key(WT_CURSOR *cursor)
         }
 
         WT_ERR(__curversion_set_value_with_format(cursor, WT_CURVERSION_METADATA_FORMAT,
-          twp->start_txn, twp->start_ts, twp->durable_start_ts, twp->stop_txn, twp->stop_ts,
+          twp->start_txn, WT_MAX(twp->start_ts, twp->start_prepare_ts),
+          WT_MAX(twp->durable_start_ts, twp->start_prepare_ts), twp->stop_txn, twp->stop_ts,
           twp->durable_stop_ts, hs_upd_type, 0, 0, WT_CURVERSION_HISTORY_STORE));
 
         version_cursor->upd_stop_txnid = twp->start_txn;
         version_cursor->upd_durable_stop_ts = twp->durable_start_ts;
-        version_cursor->upd_stop_ts = twp->start_ts;
+        version_cursor->upd_stop_ts = WT_MAX(twp->start_ts, twp->start_prepare_ts);
```

Comment:
same here

Reply:
I'll put this as a new ticket for now https://jira.mongodb.org/browse/WT-15060